joomla/joomla-cms
# [6.0] Updater: Remove Adapter classes from inheritance

### Summary of Changes
This is an effort to remove deprecated code from the system. The Updater class depends on the Adapter class, which in turn has been deprecated for a long time already. This PR removes the code of the Adapter class from the Updater class and replaces it with some sane methods to add adapters when required.

This PR does some cleanup and it adds the following methods:
-  loadAdapters(): This iterates over the set folder and adds all files it finds as available updater adapters. To keep it light, the array only contains the name as key and either the class name as value or an object. The object is instantiated just in time when needed.
-  getAdapters(): This method keeps the behavior of the $custom parameter, but otherwise still just returns the list of available adapters
-  getAdapter(): This new method replaces loadAdapter() and either returns an object which was previously made or instantiates it from the class stored in the $adapters array.
-  setAdapter(): This new method allows to hand in a name for the adapter and either a class name or an instantiated object for the respective adapter type. This would allow you to add custom adapter types and also to overwrite existing adapter types.

I reused some method names from the Adapter class. This PR is a b/c break and thus targeted towards 6.0. This is the corresponding PR to #43792 for the Updater classes.

### Testing Instructions
Install an old version of an extension and then run an update for it, both via the normal extension updater and by installing a newer package over the old one. This should work the same both before and after applying the change.


### Link to documentations
Please select:

- [X] Pull Request link for manual.joomla.org: https://github.com/joomla/Manual/pull/301
